### PR TITLE
Feat/day picker component

### PR DESCRIPTION
**Pull request by sepehrnoor**
## Explain the changes you’ve made

I have added a component for choosing a date for use in the booking part of the app. In addition, I have updated the Jest config to work with the new component (which uses a lot of native modules behind the scenes). I have also updated `styled-components` imports in a few files, which were preventing the tests from running, to the correct React Native submodule.

## Explain why these changes are made

This component is necessary for picking a date in the booking part of the app. The changes to the Jest setup and config and the import statements in the other components are necessary for running the tests for this component, and possibly other components in the future as they are commonly used modules and libraries. 

## Explain your solution

The component was created by using the node module `react-native-calendars`. The Jest configuration was done by changing the transform whitelist in the Jest config in `package.json` to include certain packages written in ES6. In addition, the Jest setup file was updated to mock several native modules that were preventing Jest from running.

## How to test the changes?

1. Checkout this branch
2. Run `yarn install` to install the new packages (`npm` not recommended due to a bug with Jest 27 that requires manual resolution as a workaround. `npm` cannot do this but `yarn` can).
3. Run the tests by running `npm run test`
4. Enable storybook mode in the `.env` file
5. Run the application in the simulator by running `npm run ios`
6. The new component can be found under "Date Picker" in the storybook navigator

## Was this feature tested in the following environments?
- [X] Storybook on a iOS device/simulator.
- [] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

### CODE REVIEW

File: source/components/molecules/DayPicker/DayPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    const isSelected = date === selectedDate;
+    markedDates[date] = {
+      selected: true,
+      customStyles: isSelected
+        ? dayStyles.selectedStyle
+        : dayStyles.availableStyle,
+    };
+  });
+
+  function selectDate(day) {
```
Comment (karlknowit) [inline source/components/molecules/DayPicker/DayPicker.tsx:94]: use arrow function:

```js
const selectDate = (day: string) => {...}
```

File: source/components/molecules/DayPicker/DayPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  onDateSelected: (dateString: string) => void;
+  startDate?: string;
+}
+
+const DayPicker: React.FC<DayPickerProps> = ({
+  availableDates,
+  onDateSelected,
+  startDate,
+}) => {
+  const [selectedDate, setSelectedDate] = useState();
```
Comment (karlknowit) [inline source/components/molecules/DayPicker/DayPicker.tsx:21]: Do we really need the selectedDate state?
I mean, if we already have a callback function called onDateSelected, we can assume that the parent rendering this calendar component also have some logic to save the selected date right?

**Review by karlknowit — COMMENTED** [review]
Looks good, only some minor things

File: source/components/molecules/DayPicker/DayPicker.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { storiesOf } from "@storybook/react-native";
+import React, { useState } from "react";
+import { View } from "react-native";
+import { H1 } from "../../atoms/Heading/Heading";
```
[the remark this answers is not in the document]
Comment (sepehrnoor) [reply source/components/molecules/DayPicker/DayPicker.stories.tsx]: Fixed, now using <Heading type="h1">.

File: source/components/molecules/DayPicker/DayPicker.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  const theme = useContext(ThemeContext);
+
+  const colors = {
```
[the remark this answers is not in the document]
Comment (sepehrnoor) [reply source/components/molecules/DayPicker/DayPicker.tsx]: Fixed, now using a separate getTheme.js file that accepts the theme objects for styling.

File: source/components/molecules/DayPicker/DayPicker.tsx (hunk 2 above)
Comment (sepehrnoor) [reply source/components/molecules/DayPicker/DayPicker.tsx]: Good catch. I have changed the component to be stateless and instead use a `selectedDate` prop from the parent.

**Comment by sepehrnoor** [comment]
Smoke tests are failing due to the old components `Card`, `SummaryList` and `FormCaseScreen` not adhering to new linting config. These are the files that have had their import statement updated in order to allow the tests to run, and are therefore included in the smoke test. I could lint the files by hand but it feels out of scope for this PR, what do you guys think?

File: source/components/molecules/DayPicker/getTheme.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+export default (theme) => {
+  const colors = {
```
Comment (karlknowit) [inline source/components/molecules/DayPicker/getTheme.js:2]: This is an object, so preferably rename it to **color** instead of **colors**

**Review by karlknowit — CHANGES_REQUESTED** [review]

**Comment by KenthLjungHBG** [comment]
> Smoke tests are failing due to the old components `Card`, `SummaryList` and `FormCaseScreen` not adhering to new linting config. These are the files that have had their import statement updated in order to allow the tests to run, and are therefore included in the smoke test. I could lint the files by hand but it feels out of scope for this PR, what do you guys think?

You can ignore them for now; we'll have to find a better strategy for dealing with existing linting errors. For now, makes sure any additions/changes adhere to the config, and apply styling to files when relevant (like if you've made major changes to a file).

**Review by karlknowit — APPROVED** [review]

**Review by Sven65 — APPROVED** [review]
Looks good, but I couldn't select any dates in the storybook "Today" component.
Pre-determined dates work though.

**Comment by sepehrnoor** [comment]
> 
> 
> Looks good, but I couldn't select any dates in the storybook "Today" component. Pre-determined dates work though.

This is expected behaviour (and bad storybook design on my part) since I didn't pass any selectable dates. The "Today" storybook component was added to test whether it could deduce the current date when not explicitly given a start date.
That being said, I should add a few selectable dates for a better test.

**Pull request merged**